pallets/identity/src/auth.rs

```diff
@@ -185,6 +185,10 @@ impl<T: Config> Module<T> {
         if let Some(expiry) = auth.expiry {
             let now = <pallet_timestamp::Pallet<T>>::get();
             ensure!(expiry > now, AuthorizationError::Expired);
+            if expiry < now {
+                <Authorizations<T>>::remove(target, auth_id);
+                <AuthorizationsGiven<T>>::remove(auth.authorized_by, auth_id);
+            }
```

Comment:
This line will throw an error if the authorization is expired, so the code to remove the expired authorization will not execute:
`ensure!(expiry > now, AuthorizationError::Expired);`

Reply:
If the authorization is expired, then the code needs to remove it, then return an error.

Also please add unit tests, to check that the error is thrown and that the expired authorization is removed.